Cargo.toml

```diff
@@ -12,10 +12,14 @@ repository = "https://github.com/snapview/tungstenite-rs"
 version = "0.13.0"
 edition = "2018"
 
+[package.metadata.docs.rs]
+features = ["native-tls"]
+
 [features]
-default = ["tls"]
-tls = ["native-tls"]
-tls-vendored = ["native-tls", "native-tls/vendored"]
+default = []
+use-native-tls = ["native-tls"]
+use-native-tls-vendored = ["use-native-tls", "native-tls/vendored"]
+use-rustls = ["rustls", "webpki", "webpki-roots"]
```

Comment:
@strohel, but are not `use-[name_of_the_crate]` not self-explanatory names of features? I mean, I don't think that `use-native-tls` or `use-rustls` would be confusing as it's basically like saying "use mylibrary1", "use mylibrary2", i.e. quite a regular English.

Reply:
Checking other prominent crates in the Rust ecosystem (e.g. tokio), the feature names seem to be formulated as simple nouns, not as verbs/sentences/orders. My main concern here is consistency (across ecosystem; in terms of naming style and also in terms of using the very same names for identical intents), i.e. reduced fragmentation, thus less cognitive load for our downstreams.

I agree that `use-something` sounds naturally in English. But extending the idea, then nearly every feature flag could be named `use-something`, i.e. the `use` would bring little information value (as in namig every struct `SomeStruct`, every enum `SomeEnum` bringing also little information value).

Another rather minor rationale is that `use-rustls` says "Use rustls for something", while `rustls-tls` says "(enable) TLS through rustls", arguably slightly more information. (this was already mentioned earlier in the debate)

Reply:
I personally prefer the Reqwest way with `rustls-tls` and `native-tls` as well. For one reason just for the consistency and as you already explained so nicely because `use-rustls` doesn't say for what it's used whereas `rustls-tls` clearly says it's used for TLS.

What is the final decision for this?

Reply:
I had a chat with @application-developer-DA, he'll survive `rustls-tls` and `native-tls` ( :-) ), please go ahead.